### PR TITLE
Create job to send info about rate limiting to Datadog

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Add a job to send info about our requests remaining to Githup API

### Motivation

Understand why we get rate limited

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/github_tasks.py [hunk 1]
```diff
@@ -165,3 +167,19 @@ def get_milestone_id(_, milestone):
...
+    gh = GithubAPI('DataDog/datadog-agent')
+    print(gh.get_rate_limit_info())
+    metric = create_count(
+        metric_name='github.rate_limit.remaining',
+        timestamp=int(time.time()),
+        value=gh.get_rate_limit_info()[0],
+        tags=['source:github', 'repository:datadog-agent', f'pipeline_id:{pipeline_id}'],
+    )
+    send_metrics([metric])
+    # gh.send_rate_limit_info_datadog()
```
Comment (spencergilbert) [inline tasks/github_tasks.py:185]: Left in, commented out?

Comment (KevinFairise2) [reply tasks/github_tasks.py]: removed

File: tasks/github_tasks.py [hunk 2]
```diff
@@ -165,3 +167,19 @@ def get_milestone_id(_, milestone):
...
+
+@task
+def send_rate_limit_info_datadog(_, pipeline_id):
+    from .libs.common.github_api import GithubAPI
+
+    gh = GithubAPI('DataDog/datadog-agent')
+    rate_limit_info = gh.get_rate_limit_info()
+    print(f"Remaining rate limit: {rate_limit_info[0]}/{rate_limit_info[1]}")
+    metric = create_count(
+        metric_name='github.rate_limit.remaining',
```
Comment (chouquette) [inline tasks/github_tasks.py:180]: Isn't this the number of request we already used?
ie. shouldn't the remaining number of requests be `rate_limit_info[1] - rate_limit_info[0]` ?

Comment (KevinFairise2) [reply tasks/github_tasks.py:180]: `get_rate_limit()` returns a tuple with `(requests remaining, request limit)` . So I think this what we want

Comment (chouquette) [reply tasks/github_tasks.py:180]: Then all good :)

**Review by chouquette — APPROVED** [review]

**Pull request merged**